### PR TITLE
nodejs week3

### PR DESCRIPTION
**Pull request by Infamou5**
Find homework in the meal-sharing repo.

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -57,20 +57,52 @@ router.post("/", async (request, response) => {
...
+      .andWhere("title", "like", `%${title}%`)
+      .andWhere("meal.created_date", ">", createdAfter)
+      .join("reservation", "meal.id", "=", "reservation.meal_id")
+      .where("max_reservations", ">", `${availableReservations ? "number_of_guests" : 0}`)
+      .limit(`${limit}`);
+    res.status(200).json(result);
+  } catch (error) {
+    res.status(400).json(error);
+  }
+});
```
Comment (dsamourkasidis) [inline src/backend/api/meals.js:105]: Very nice thinking!
Two comments though
- Don't forget to parse integers/floats and optionally convert string variables to lowercase
- You are potentially adding some additional overhead to your database when querying on unnecessary columns. To make it absolutely perfect if you want to try this:
```
.where((builder) => {

    if (req.query.customerId)
        builder.where('wo.customerId', req.query.customerId);

    if (req.query.companyId)
        builder.where('wo.companyId', req.query.companyId);

    if (req.query.invoiceId)
        builder.where('wo.invoiceId', req.query.invoiceId);

    if (req.query.open)
        builder.whereNull('wo.finishedAt');

    // so on..

});
```
https://stackoverflow.com/questions/54274014/how-to-create-where-conditions-dynamically-in-knex-js

File: src/backend/api/reservations.js [hunk 2]
```diff
@@ -3,21 +3,51 @@ const router = express.Router();
...
+    let errorMessage = "";
+
+    if (isNaN(request.body.number_of_guests)) {
+      requestValid = false;
+      errorMessage += "Number of guests is not a number.";
+    }
+
+    if (!isNaN(request.body.contact_name)) {
+      requestValid = false;
+      errorMessage += "Contact is a number. Should be string.";
```
Comment (dsamourkasidis) [inline src/backend/api/reservations.js:30]: Again well done on considering validation of the incoming data!
To make it even more clean and nice, consider create a different function eg validateReservation

**Review by dsamourkasidis — APPROVED** [review]
Awesome work :+1: 
Everything is super good!
I just left small comments for you to have in mind and if you want to make things even more perfect, but they're absolutely optional you're good to go!

File: src/backend/api/meals.js (hunk 1 above)
Comment (Infamou5) [reply src/backend/api/meals.js:105]: Thank you for the link!

File: src/backend/api/reservations.js (hunk 2 above)
Comment (Infamou5) [reply src/backend/api/reservations.js:30]: Ran out of time. Otherwise I would have done that too.

**Pull request merged**